platform/src/main/java/org/stellar/anchor/platform/controller/HealthController.java

```diff
@@ -16,10 +19,20 @@ public class HealthController {
   }
 
   @RequestMapping(method = {RequestMethod.GET})
-  public HealthCheckResponse health(@RequestParam(required = false) List<String> checks) {
+  public ResponseEntity<HealthCheckResponse> health(
+      @RequestParam(required = false) List<String> checks) {
     if (checks == null) {
       checks = List.of("all");
     }
-    return healthCheckService.check(checks);
+    HealthCheckResponse healthCheckResponse = healthCheckService.check(checks);
+
+    boolean unhealthy =
+        healthCheckResponse.getChecks().values().stream()
+            .anyMatch(result -> result.getStatus() == HealthCheckStatus.RED);
+    if (unhealthy) {
+      return new ResponseEntity<>(healthCheckResponse, HttpStatus.INTERNAL_SERVER_ERROR);
+    } else {
+      return ResponseEntity.ok(healthCheckResponse);
+    }
```

Comment:
With this setup you won't be able to identify what exactly is wrong, making most of the health check classes infrastructure useless.

Reply:
Good to see you here, Marcelo!

This only marks the failure of the health check. The detail/reason of the failure is still in the JSON response.